tests/acceptance/routing-test.js

```diff
@@ -12,51 +12,147 @@ module('Acceptance | routing', {
+  },
+  {
+    filename: "index.template.hbs",
+    content: "<p>Main Page</p>"
+  },
+  {
+    filename: "application.template.hbs",
+    content: `{{#link-to "index" class="test-index-link"}}Index{{/link-to}}
+              {{#link-to "about" class="test-about-link"}}About{{/link-to}}
+
+              {{outlet}}`
+  },
+  {
+    filename: "router.js",
+    content: `import Ember from 'ember';
+              import config from './config/environment';
+
+              var Router = Ember.Router.extend({
+                location: config.locationType
+              });
+
+              Router.map(function() {
+                this.route("about");
+              });
 
-  const files = [
-    {
-      filename: "about.template.hbs",
-      content: "<p>About Page</p>"
-    },
-    {
-      filename: "index.template.hbs",
-      content: "<p>Main Page</p>"
-    },
-    {
-      filename: "application.template.hbs",
-      content: "{{#link-to \"index\" class=\"test-index-link\"}}Index{{/link-to}}\n{{#link-to \"about\" class=\"test-about-link\"}}About{{/link-to}}\n\n{{outlet}}"
-    },
-    {
-      filename: "router.js",
-      content: "import Ember from 'ember';\nimport config from './config/environment';\n\nvar Router = Ember.Router.extend({\n  location: config.locationType\n});\n\nRouter.map(function() {\n  this.route(\"about\");\n});\n\nexport default Router;\n"
-    },
-    {
-      filename: "twiddle.json",
-      content: "{\n  \"version\": \"0.4.0\",\n  \"dependencies\": {\n    \"jquery\": \"https://cdnjs.cloudflare.com/ajax/libs/jquery/1.11.3/jquery.js\",\n    \"ember\": \"https://cdnjs.cloudflare.com/ajax/libs/ember.js/1.13.10/ember.js\",\n    \"ember-data\": \"https://cdnjs.cloudflare.com/ajax/libs/ember-data.js/1.13.13/ember-data.js\",\n    \"ember-template-compiler\": \"https://cdnjs.cloudflare.com/ajax/libs/ember.js/1.13.10/ember-template-compiler.js\"\n  }\n}"
-    }
-  ];
-
-  const aboutLink = '.test-about-link';
-  const indexLink = '.test-index-link';
-  const outletText = 'p';
+              export default Router;`
+  },
+  {
+    filename: "twiddle.json",
+    content: `{
+                "version": "0.4.0",
+                "dependencies": {
+                  "jquery": "https://cdnjs.cloudflare.com/ajax/libs/jquery/1.11.3/jquery.js",
+                  "ember": "https://cdnjs.cloudflare.com/ajax/libs/ember.js/1.13.10/ember.js",
+                  "ember-data": "https://cdnjs.cloudflare.com/ajax/libs/ember-data.js/1.13.13/ember-data.js",
+                  "ember-template-compiler": "https://cdnjs.cloudflare.com/ajax/libs/ember.js/1.13.10/ember-template-compiler.js"
+                }
+              }`
+  }
+];
+
+test('Able to do routing in a gist', function(assert) {
   let iframe_window;
 
-  runGist(files);
+  runGist(TWIDDLE_WITH_ROUTES);
 
   andThen(function() {
-    iframe_window = outputPane();
+    const done = assert.async();
+
+    return new Ember.RSVP.Promise(function (resolve) {
+      Ember.run.next(function() {
```

Comment:
rather then the `next` here, which may be racy, we could make the async communication with the remote frame part of the test-waiters. This would make `andThen` "just work"